ftw/tokenauth/oauth2/browser/oauth2_token.py

```diff
@@ -284,6 +283,7 @@ def verified_subject(self, claimset, service_key):
                     "JWT subject doesn't match user_id of service key.")))
         return subject
 
+    def issue_access_token(self, service_key, user_id):
```

Comment:
I would consider naming this `subject` instead of `user_id`, all the way down the chain. The name `user_id` doesn't carry that much meaning by itself, except that it's the ID of *some* user.

What's more interesting in the context where this shows up (token metadata, PAS plugin, usage logs, ...) is whether it's the **actor** or the **subject**.

It's a tedious change, I know, but I think there's value in clearly separating these two concepts. There should never be any ambiguity at all about which is which, and it would be helpful if the code would express that, where possible.

Reply:
Hmm, I'm not sure about that. `Subject` is the term used in JWT in the context of a bearer token, while `user_id` is a more specific name for what it is used for in ftw.tokenauth. I see some value in making it visible that a userid is expected here.

I see the point about distinguishing between actor and subject, but that's already done where needed (in the `verified_subject` method).

In the context of Plone and it's authentication system the term subject is not common. Thus displaying subject in the logs and actor in the service key properties would be more confusing for me.

Reply:
Fair enough, I'm fine with leaving it as-is. I might add a few clarifying comments on my next pass though ;)